server/sonar-server/src/main/java/org/sonar/server/component/index/ComponentIndexQuery.java

```diff
@@ -35,6 +37,7 @@
 
   private String query;
   private Collection<String> qualifiers = Collections.emptyList();
+  private Set<String> componentUuids = Collections.emptySet();
```

Comment:
I think I would prefer `Optional<Set<String>>`, with the default set to `Optional.empty()`. This would make the empty set a valid case, which would not return any results.